flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala

```diff
@@ -417,12 +452,33 @@ class Table(
 
   private def join(right: Table, joinPredicate: Option[Expression], joinType: JoinType): Table = {
     // check that right table belongs to the same TableEnvironment
```

Comment:
// check that  the TableEnvironment of right table is not null and right table belongs to the same TableEnvironment